parachain/src/primitives.rs

```diff
@@ -259,7 +259,7 @@ pub type UpwardMessage = Vec<u8>;
 
 /// Validation parameters for evaluating the parachain validity function.
 // TODO: balance downloads (https://github.com/paritytech/polkadot/issues/220)
-#[derive(PartialEq, Eq, Decode)]
+#[derive(PartialEq, Eq, Decode, Clone)]
```

Comment:
why not put this derive under `cfg_attr(feature = "std"` or `cfg_attr(test`?

Reply:
Heh, that also crossed my mind: decided to go with this because that would be artificially limiting, a bit more code and having Clone (and in other places other derives) do not harm.

An argument for adding a conditional derive is to show that these traits are used in tests for now. But I didn't think this is an important enough.

I am not against changing this though if you think that's a good thing.

Reply:
No strong opinion either, just seems a bit inconsistent to feature-gate some derives like `Encode` here, but not others.